### PR TITLE
perf(moe): add gfx1250 fused sigmoid top-k

### PR DESCRIPTION
**Pull request by Yu-Zhewen**
## Summary
- Port the gfx950 fused sigmoid-bias top-k route from #954 to gfx1250 for the Kimi-K3 896-expert / top-16 contract.
- Keep decode M=1 on the existing Triton specialist; Gluon covers batched prefill tokens.
- Merge the gfx950 and gfx1250 AMD tests into one file.

## Performance
Kernel latency vs. main on gfx1250 (`E=896, topk=16`).

| M | Main (µs) | PR (µs) | Speedup vs. main |
|--:|----------:|--------:|-----------------:|
| 1 | 33.15 | 23.90 | 1.39x |
| 2 | 34.55 | 23.82 | 1.45x |
| 4 | 34.19 | 23.94 | 1.43x |
| 8 | 34.21 | 23.84 | 1.44x |

### CODE REVIEW

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/sigmoid_topk.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def _gluon_eligible(
         and topk <= experts <= 1024
         and router_logits.stride(1) == 1
         and correction_bias.is_contiguous()
+        and correction_bias.dtype == router_logits.dtype
```
[the remark this answers is not in the document]
Comment (Yu-Zhewen) [reply tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/sigmoid_topk.py]: Agreed. Matching-dtype is no longer part of shared `_gluon_eligible`. gfx1250 stays float32 via its kernel/registry signatures; gfx950 mixed-dtype routes remain eligible.

File: tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/gfx1250/moe/mxfp4/routing.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
+# Copyright (c) 2026 LightSeek Foundation
```
[the remark this answers is not in the document]
Comment (Yu-Zhewen) [reply tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/gfx1250/moe/mxfp4/routing.py:1]: Leaving the abbreviated copyright. That matches the new gfx1250 files in this series and several existing gfx1250 modules.

File: tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/gfx1250/moe/mxfp4/routing.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    )
+
+
+def invoke_sigmoid_bias_topk_route_prefill_gluon(
+    router_logits: torch.Tensor,
+    correction_bias: torch.Tensor,
+    topk: int,
+    *,
+    routed_scaling_factor: float = 1.0,
+    normalize_topk_weights: bool = True,
```
[the remark this answers is not in the document]
Comment (Yu-Zhewen) [reply tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/gfx1250/moe/mxfp4/routing.py:114]: Leaving the invoke defaults. They match the gfx950 `invoke_sigmoid_bias_topk_route_prefill_gluon` contract this port copies.

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/gluon/sigmoid_topk.py [hunk 4]
```diff
@@ -70,12 +68,48 @@ def gluon_sigmoid_bias_topk_gfx950(
...
+    def gluon_sigmoid_bias_topk_gfx1250(
+        *,
+        router_logits: torch.Tensor,
+        correction_bias: torch.Tensor,
+        topk: int,
+        routed_scaling_factor: float,
+        normalize_topk_weights: bool,
+        weights_dtype: torch.dtype = torch.float32,
+    ) -> tuple[torch.Tensor, torch.Tensor]:
+        topk_ids, topk_weights = invoke_sigmoid_bias_topk_route_prefill_gfx1250(
```
[the remark this answers is not in the document]
Comment (Yu-Zhewen) [reply tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/gluon/sigmoid_topk.py:99]: Leaving the wrapper without a new docstring. It is a thin registry shim over the documented gfx1250 invoke helper.

File: tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/gfx1250/moe/mxfp4/routing.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    topk_lane = gl.arange(0, TKP, layout=topk_layout)
+    selected_ids = gl.zeros([TKP], gl.int32, topk_layout)
+    selected_weights = gl.zeros([TKP], gl.float32, topk_layout)
+    sentinel = gl.full([EP], E, gl.int32, expert_layout)
+    for rank in gl.static_range(TOPK):
+        maximum = gl.max(choice, axis=0)
+        selected = gl.min(
+            gl.where((choice == maximum) & expert_mask, expert, sentinel),
+            axis=0,
+        )
```
[the remark this answers is not in the document]
Comment (Yu-Zhewen) [reply tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/gfx1250/moe/mxfp4/routing.py:77]: Leaving the sentinel as `E`, same as the gfx950 kernel from #954. Graph-padding NaN handling is unchanged from that port.

File: tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/gfx1250/moe/mxfp4/routing.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    routed_scaling_factor: float = 1.0,
+    normalize_topk_weights: bool = True,
+) -> tuple[torch.Tensor, torch.Tensor]:
+    """Apply Kimi's noaux_tc route with one independent CTA per token."""
+    if (
+        router_logits.ndim != 2
+        or router_logits.dtype not in _ROUTE_DTYPES
+        or not router_logits.is_cuda
+        or not 0 < topk <= 16
+        or router_logits.shape[1] > 1024
```
[the remark this answers is not in the document]
Comment (Yu-Zhewen) [reply tokenspeed-kernel-amd/python/tokenspeed_kernel_amd/ops/gfx1250/moe/mxfp4/routing.py:122]: The public `moe_sigmoid_bias_topk` already requires `topk <= experts` before this backend runs. Direct backend calls are the same shape contract as gfx950.
